### PR TITLE
remove the "building the docs" locally instructions

### LINKED ISSUE #627: Local docfx build is now much less useful
Note: I realize this is not going to be high priority right now, so feel free to ignore this for the moment.

Until recently, running docfx locally on this repo created a full site pretty much equivalent to http://dotnet.github.io/, that was useful for checking pretty much anything about the documentation.

But now, presumably due to the move to docs.microsoft.com, running docfx produces just the bare article body, with no TOC or any other UI elements.

Out of curiosity, I also tried running `.openpublishing.build.ps1`, but that failed when accessing https://github.com/Microsoft/templates.docs.msft.

### LINKED ISSUE #810: Sample directory doesn't quite mirror the proposed guidance
The directories under /samples should mirror the directories under /docs. 

They don't now for historical reasons.

The concern with moving all the samples quickly is that the sample code is referenced from many different md files within the repository. We need to update all those links when any file is moved.

### PR DESCRIPTION
**Pull request by BillWagner**
Fixes #627
Fixes #810 

Cleaning old issues, and these two are in the same file.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -105,11 +105,9 @@ Code all lives in the [dotnet/samples](https://github.com/dotnet/samples) reposi
 
 - *snippets* contains the code snippets used throughout the documentation. Inside the *snippets* folder, the folder names identify the language of the code snippet.
 
-We will expand these directories as the docs repository adds new content. For example, we will add Xamarin directories, like `xamarin-ios` and `xamarin-android` directories.
+Under these top-level folders, the samples structure mirrors the folder structure of the related articles. For example, **core/get-started** contains the samples that relate to the articles in the **docs/core/get-started** folder. Snippets for the C# Language keywords are in **snippets/csharp/reference/keywords**.
```
Comment (pkulikov) [inline CONTRIBUTING.md]: @BillWagner those are not the best examples:
- There is no **core/get-started** folder. It's [core/getting-started](https://github.com/dotnet/samples/tree/master/core/getting-started)
- There is no **docs/core/get-started** folder. There is [get-started.md file](https://github.com/dotnet/docs/blob/master/docs/core/get-started.md). However, above mentioned samples look to serve [docs/core/testing](https://github.com/dotnet/docs/tree/master/docs/core/testing) folder
- Snippets for the C# keywords are all over the samples repo. For example, check the snippet location for the [ascending.md file](https://raw.githubusercontent.com/dotnet/docs/master/docs/csharp/language-reference/keywords/ascending.md). Though, I see and follow the tendency to place new C# keyword snippets to the [snippets/csharp/keywords](https://github.com/dotnet/samples/tree/master/snippets/csharp/keywords) folder (I guess, we forgot about the "language-reference" subfolder when created that project)

So, what would be a good example? Something recently created. For example, [machine-learning/tutorials](https://github.com/dotnet/samples/tree/master/machine-learning/tutorials) in the **samples** repo support [docs/machine-learning/tutorials](https://github.com/dotnet/docs/tree/master/docs/machine-learning/tutorials) in the **docs** repo.

As for snippets, there is [snippets/csharp/how-to](https://github.com/dotnet/samples/tree/master/snippets/csharp/how-to) and [docs/csharp/how-to](https://github.com/dotnet/docs/tree/master/docs/csharp/how-to) pair.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -105,11 +105,9 @@ Code all lives in the [dotnet/samples](https://github.com/dotnet/samples) reposi
 
 - *snippets* contains the code snippets used throughout the documentation. Inside the *snippets* folder, the folder names identify the language of the code snippet.
```
Comment (mairaw) [inline CONTRIBUTING.md]: it seems that we were also adopting the path `https://github.com/dotnet/samples/tree/master/snippets/<language>/api/<namespace>/<apiname>` for api reference snippets as you can see at https://github.com/dotnet/samples/tree/master/snippets/csharp/api

also any suggestions for folder name when we're creating cross-plat samples that will go under the standard folder? They will be .NET Core SDK-style projects.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -105,11 +105,9 @@ Code all lives in the [dotnet/samples](https://github.com/dotnet/samples) reposi
...
 
-We will expand these directories as the docs repository adds new content. For example, we will add Xamarin directories, like `xamarin-ios` and `xamarin-android` directories.
+Under these top-level folders, the samples structure mirrors the folder structure of the related articles. For example, **core/get-started** contains the samples that relate to the articles in the **docs/core/get-started** folder. Snippets for the C# Language keywords are in **snippets/csharp/reference/keywords**.
 
-Sample code may fit more than one of these areas. In those cases, place the sample so it matches
-the topics you are covering in your documents. Ask yourself what readers will learn from reading
-your topic. What will they learn from building and running your sample?
+We will expand these directories as the docs repository adds new content. For example, we will add Xamarin directories, like `xamarin-ios` and `xamarin-android` directories.
```
Comment (mairaw) [inline CONTRIBUTING.md]: We will or we might?

**Review by mairaw — APPROVED** [review]
LGTM but left some comments/questions

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -160,43 +148,6 @@ The following list shows some guiding rules that you should keep in mind when yo
 
 > Note: you might notice that some of the topics are not currently following all the guidelines specified here and on the [style guide](./styleguide/template.md) as well. We're working towards achieving consistency throughout the site. Check the list of [open issues](https://github.com/dotnet/docs/issues?q=is%3Aissue+is%3Aopen+label%3Aguidelines-adherence) we're currently tracking for that specific goal.
 
-## Building the docs
```
Comment (scottaddie) [inline CONTRIBUTING.md]: What if my preferred workflow involves testing changes locally before pushing? Are we discouraging that practice?

Comment (BillWagner) [reply CONTRIBUTING.md]: At this time, yes. There are a lot of issues with the local build: You need to download the dependent repos, and have them in the right location. Xref doesn't work reliably for most community members. #627 has some of the details.

**Pull request merged**